src/ore/src/iter.rs

```diff
@@ -41,10 +41,36 @@ where
             Some(v1) => self.all(|v2| v1 == v2),
         }
     }
+
+    /// Converts the the iterator into an `ExactSizeIterator` reporting the given size.
+    fn exact_size(self, len: usize) -> ExactSize<Self> {
+        ExactSize { inner: self, len }
+    }
 }
 
 impl<I> IteratorExt for I where I: Iterator {}
 
+/// Iterator type returned by [`IteratorExt::exact_size`].
+#[derive(Debug)]
+pub struct ExactSize<I> {
+    inner: I,
+    len: usize,
+}
+
+impl<I: Iterator> Iterator for ExactSize<I> {
+    type Item = I::Item;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.inner.next()
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
```

Comment:
I think the `size_hint` function is not correct after calling `next` because it doesn't update `self.len`.

Reply:
🤦 I'm glad you caught that!